bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/action/RoomActions.java

```diff
@@ -61,90 +61,82 @@ public void setThingHandler(@Nullable ThingHandler handler) {
-            @ActionInput(name = "endtime", label = "@text/actionInputEndtimeLabel", description = "@text/actionInputEndtimeDesc") @Nullable Long endTime,
-            @ActionInput(name = "mode", label = "@text/actionInputModeLabel", description = "@text/actionInputModeDesc") @Nullable String mode) {
         CommonInterface roomHandler = handler;
-        if (roomHandler != null) {
-            String roomId = roomHandler.getId();
-            SetpointMode targetMode = SetpointMode.UNKNOWN;
-            Long targetEndTime = endTime;
-            Double targetTemp = temp;
-            if (mode != null) {
-                try {
-                    targetMode = SetpointMode.valueOf(mode);
-                    if (!ALLOWED_MODES.contains(targetMode)) {
-                        logger.info("Mode can only be MAX, HOME or MANUAL for a room");
-                        return;
-                    }
-                } catch (IllegalArgumentException e) {
-                    logger.info("Invalid mode passed : {} - {}", mode, e.getMessage());
-                    return;
-                }
-            }
-            if (temp != null) {
-                logger.debug("Temperature provided, mode forced to MANUAL.");
-                targetMode = SetpointMode.MANUAL;
-                if (targetEndTime == null) {
-                    logger.info("Temperature provided but no endtime given, action ignored");
-                    return;
-                }
-            } else {
-                if (SetpointMode.HOME.equals(targetMode)) {
-                    targetEndTime = 0L;
-                    targetTemp = 0.0;
-                } else {
-                    logger.info("mode is required if no temperature setpoint provided");
-                    return;
-                }
-            }
+        if (roomHandler == null) {
+            logger.info("Handler not set for room thing actions.");
+            return;
+        } else if (mode == null) {
+            logger.info("Mode is required, action ignored");
+            return;
+        }
 
-            try {
-                double setpointTemp = targetTemp != null ? targetTemp : 0;
-                long setpointEnd = targetEndTime;
-                SetpointMode setpointMode = targetMode;
-                energy.ifPresent(cap -> cap.setRoomThermTemp(roomId, setpointTemp, setpointEnd, setpointMode));
-            } catch (IllegalArgumentException e) {
-                logger.debug("Ignoring setRoomThermpoint command due to illegal argument exception: {}",
-                        e.getMessage());
+        SetpointMode targetMode = SetpointMode.UNKNOWN;
+        try {
+            targetMode = SetpointMode.valueOf(mode);
+            if (!ALLOWED_MODES.contains(targetMode)) {
+                logger.info("Mode can only be MAX, HOME or MANUAL for a room");
```

Comment:
It would be preferable to build this string from the enum names. This way it is ensured that the message is still correct if they would be renamed.

Reply:
Ok changed.